zeebe/exporters/rdbms-exporter/src/main/java/io/camunda/exporter/rdbms/handlers/ProcessInstanceExportHandler.java

```diff
@@ -75,6 +87,46 @@ private ProcessInstanceDbModel map(final Record<ProcessInstanceRecordValue> reco
         .parentElementInstanceKey(value.getParentElementInstanceKey())
         .version(value.getVersion())
         .partitionId(record.getPartitionId())
+        .treePath(createTreePath(record).toString())
         .build();
   }
+
+  public TreePath createTreePath(final Record<ProcessInstanceRecordValue> record) {
+    final var value = record.getValue();
+    final var elementInstancePath = value.getElementInstancePath();
+    final var processDefinitionPath = value.getProcessDefinitionPath();
+    final var callingElementPath = value.getCallingElementPath();
+    final Long processInstanceKey = value.getProcessInstanceKey();
+    if (elementInstancePath == null || elementInstancePath.isEmpty()) {
+      return new TreePath().startTreePath(processInstanceKey);
+    }
+
+    final TreePath treePath = new TreePath();
+    for (int i = 0; i < elementInstancePath.size(); i++) {
+      final List<Long> keysWithinOnePI = elementInstancePath.get(i);
+      treePath.appendProcessInstance(keysWithinOnePI.getFirst());
+      if (keysWithinOnePI.getFirst().equals(processInstanceKey)) {
+        // we reached the leaf of the tree path, when we reached current processInstanceKey
+        break;
+      }
+      final var callActivity =
+          ProcessCacheUtil.getCallActivityId(
```

Comment:
🔧 I think we could improve this by loading all required processes at once and not in a loop. The cache can also implement a `loadAll` method, where you query them with an `IN` filter. See [ProcessCache](https://github.com/camunda/camunda/blob/main/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/cache/ProcessCache.java#L111) in REST Gateway.

What do you think?
It can be a follow-up issue if you want to merge this already. But I think it could make sense, less DB queries.